### PR TITLE
Document `set_conversation_response` script action

### PR DESCRIPTION
**Pull request by emontnemery**
## Proposed change

Document `set_conversation_response` script action

Parent PR: https://github.com/home-assistant/core/pull/108233

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: 
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [ ] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_docs/scripts.markdown [hunk 1]
```diff
@@ -914,6 +914,32 @@ script:
           entity_id: light.ceiling
 ```
 
+## Respond to a conversation
+
+The `set_conversation_response` script action allows returning a custom response
+when an automation is triggered by a conversation engine, for example a voice
+assistant. The conversation response can be templated.
+
+```yaml
```
Comment (emontnemery) [inline source/_docs/scripts.markdown:925]:
```suggestion
{% raw %}

```yaml
```

File: source/_docs/scripts.markdown [hunk 2]
```diff
@@ -914,6 +914,32 @@ script:
...
+# Example of a templated conversation response resulting in "Testing 123"
+- variables:
+    my_var: "123"
+- set_conversation_response: "{{ 'Testing ' + my_var }}":
+```
```
Comment (emontnemery) [inline source/_docs/scripts.markdown:930]:
```suggestion
```

{% endraw %}
```

File: source/_docs/scripts.markdown [hunk 3]
```diff
@@ -914,6 +914,32 @@ script:
...
+
+The response is handed to the conversation engine when the automation finishes. If
+the `set_conversation_response` is executed multiple times, the most recent
+response will be handed to the conversation engine. To clear the response, set it
+to `None`:
+```yaml
```
Comment (emontnemery) [inline source/_docs/scripts.markdown:939]:
```suggestion

```yaml
```

File: source/_docs/scripts.markdown [hunk 4]
```diff
@@ -914,6 +914,32 @@ script:
...
+# Example of a clearing a conversation response
+set_conversation_response: ~
+```
+
```
Comment (emontnemery) [inline source/_docs/scripts.markdown:943]:
```suggestion
```

File: source/_docs/scripts.markdown [hunk 5]
```diff
@@ -914,6 +914,33 @@ script:
...
+assistant. The conversation response can be templated.
+
+{% raw %}
+
+```yaml
+# Example of a templated conversation response resulting in "Testing 123"
+- variables:
+    my_var: "123"
+- set_conversation_response: "{{ 'Testing ' + my_var }}":
+
```
Comment (c0ffeeca7) [inline source/_docs/scripts.markdown:933]:
```suggestion

{% endraw %}

```

File: source/_docs/scripts.markdown [hunk 6]
```diff
@@ -914,6 +914,35 @@ script:
...
+when an automation is triggered by a conversation engine, for example a voice
+assistant. The conversation response can be templated.
+
+{% raw %}
+
+```yaml
+# Example of a templated conversation response resulting in "Testing 123"
+- variables:
+    my_var: "123"
+- set_conversation_response: "{{ 'Testing ' + my_var }}":
```
Comment (c0ffeeca7) [inline source/_docs/scripts.markdown:929]:
```suggestion
- set_conversation_response: "{{ 'Testing ' + my_var }}":
```
```

**Review by c0ffeeca7 — APPROVED** [review]
LGTM :+1:

**Pull request merged**